Add Github workflows for CI and Publish

SiimMardus opened the pull request:
Currently publishing was done manually. The publish workflow now
builds the dist folder and publishes it to NPM registry on changes
to master branch.

The CI workflow runs linter and tests on PR updates.

Followed this guide: https://github.com/marketplace/actions/npm-publish#github-action
Took inspiration from: https://github.com/salemove/freddy/tree/master/.github/workflows

SiimMardus commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,19 @@
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+      - run: npm ci
```
nvm, we dont have that command

SiimMardus commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,19 @@
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "20"
+      - run: npm lint
+      - run: npm test
```
I put those two here as they made most sense to check.

`npm build` has so far been run by the developer, so I don't think this is necessary here? Although it would ensure that the version sent to npm will have the correct built files, it could also make it confusing in case of a push to master without updating dist, because the master branch and npm package would have different builds in the dist 🤔

indrekj replied:
Oh, We're using [travis](https://github.com/salemove/client-logger-js/blob/master/.travis.yml). This doesn't work. Could you delete the travis.

Then please also add `ci.yml` and run the linter and tests there ([example](https://github.com/salemove/freddy/blob/master/.github/workflows/ci.yml)). The publish.yml can be kept only for the release (run on master).

The test should ensure that the dist is up to date but not generate a new one (unless we only generate the dist during release process and not check it in at all).

indrekj replied:
I added NPM_TOKEN to the repository secrets now as well.

SiimMardus replied:
I think it would make sense to gitignore the dist folder and only generate it during release.

indrekj replied:
That is an option yes. I think I initially added to quickly see how the bundle size changes. But imo that's a different problem and doesn't have to be solved like that.

SiimMardus replied:
[Remove travis](https://github.com/salemove/client-logger-js/pull/21/commits/24d0648a851617444cf564d042e916c1127287ce) and [Add github workflows for CI and Publish](https://github.com/salemove/client-logger-js/pull/21/commits/a8283677b80019aa58005a75f4e03a998787d27c)

indrekj commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,21 @@
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
```
if you're running against one version (whuich you do accoring to line 19), then you can just remove the stategy part

SiimMardus replied:
[force-pushed](https://github.com/salemove/client-logger-js/compare/c3baa8003747bf2c123966b1cfed0a06ea7a5a1b..7a1495844db30680d740e2ed94aaf817100e3c3b)

indrekj commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,18 @@
+
+on:
+  push:
+    branches: master
```
`branches: [master]`
I think it should be an array, maybe string works as well but not sure

SiimMardus replied:
Took from here https://github.com/marketplace/actions/npm-publish#github-action

SiimMardus replied:
Github seems to understand the branch though, as the actions are run (although fail with other syntax error) https://github.com/salemove/client-logger-js/actions/workflows/ci.yml

So this part seems fine

SiimMardus replied:
Will actually unify ci.yml also, since this was copied from freddy and remade.

indrekj replied:
I would unify under array tbh, beacuse they key is plural

SiimMardus replied:
[force-pushed](https://github.com/salemove/client-logger-js/compare/f9b3d13b6411fde5db35a24a9f99c95e191f44bd..0190656063fe15fc4ccb24e2a2eaf8a107d647b2) ✅

indrekj commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,18 @@
+    branches: master
+
+jobs:
+  test:
```
it says there's a syntax errors on line 10 but I don't see it https://github.com/salemove/client-logger-js/actions/runs/7640500317/workflow

SiimMardus replied:
VSCode GH actions linter extension saved the day [force-pushed](https://github.com/salemove/client-logger-js/compare/f9b3d13b6411fde5db35a24a9f99c95e191f44bd..0190656063fe15fc4ccb24e2a2eaf8a107d647b2) at least the action ran now. Proceeding to fix other stuff

indrekj reviewed (approved):
nice

SiimMardus commented:
Previously tests failed because mocha (and all deps for that matter were missing), added it to ci.yml. 

Also changed it in publish.yml to ensure that the action has `rollup` to even build it.

SiimMardus commented:
Let's hope this works! 🤞 🐐 🔪 🩸 🐐 🔪 🩸 🐐 🔪 🩸 🐐 🔪 🩸

SiimMardus commented:
!merge
